rmf_site_editor/src/interaction/select/place_object.rs

```diff
@@ -31,38 +34,28 @@ impl Plugin for ObjectPlacementPlugin {
 #[derive(Resource, Clone, Copy)]
 pub struct ObjectPlacementServices {
     pub place_object_2d: Service<Option<Entity>, ()>,
-    pub place_object_3d: Service<Option<Entity>, ()>,
-    pub replace_parent_3d: Service<Option<Entity>, ()>,
-    pub hover_service_object_3d: Service<(), (), Hover>,
 }
 
 impl ObjectPlacementServices {
     pub fn from_app(app: &mut App) -> Self {
-        let hover_service_object_3d = app.spawn_continuous_service(
-            Update,
-            hover_service::<PlaceObject3dFilter>
-                .configure(|config: SystemConfigs| config.in_set(SelectionServiceStages::Hover)),
-        );
         let place_object_2d = spawn_place_object_2d_workflow(app);
-        let place_object_3d = spawn_place_object_3d_workflow(hover_service_object_3d, app);
-        let replace_parent_3d = spawn_replace_parent_3d_workflow(hover_service_object_3d, app);
-        Self {
-            place_object_2d,
-            place_object_3d,
-            replace_parent_3d,
-            hover_service_object_3d,
-        }
+        Self { place_object_2d }
     }
 }
 
 #[derive(SystemParam)]
 pub struct ObjectPlacement<'w, 's> {
     pub services: Res<'w, ObjectPlacementServices>,
     pub commands: Commands<'w, 's>,
+    current_level: Res<'w, CurrentLevel>,
 }
 
 impl<'w, 's> ObjectPlacement<'w, 's> {
-    pub fn place_object_2d(&mut self, object: Model, level: Entity) {
+    pub fn place_object_2d(&mut self, object: Model) {
+        let Some(level) = self.current_level.0 else {
+            warn!("Unble to create [object:?] outside a level");
+            return;
+        };
```

Comment:
This is part of my effort to reduce code duplication. I refactored both 2d and 3d object placement to take the same input (just a `Model`).
Specifically for the fuel widget, it was really the same between this app and the workcell editor, with the only different that the spawning function was different. I introduced a field in the plugin that allows users to pass a spawning function that takes as an input `Commands` and a `Model`, and will be called when the `Spawn model` button is clicked. This allowed reusing the code between the two applications but comes with the drawback that this minor API change is necessary.

https://github.com/open-rmf/rmf_site/blob/3496425691f9c43b46b6acb59bd257961e80a6f9/rmf_site_editor/src/widgets/fuel_asset_browser.rs#L28-L40